mp34dt01_l4_Driver.c

```diff
@@ -0,0 +1,904 @@
+*/
+uint8_t MP34DT01_Pause(void)
+{
+  int32_t counter = 0;
+  for (counter = MP34DT01_Handler.MicChannels; counter > 0; counter --)
+  {
+    if (HAL_OK != HAL_DFSDM_FilterRegularStop_DMA(&haudio_in_dfsdmfilter[counter-1]))
+    {
+      return AUDIO_ERROR;
+    }
+  }
+  /* Return 0 if all operations are OK */
+  return AUDIO_OK;
+}
+
+/**
+* @brief  Resumes the audio file stream.
+* @param  None
+* @retval AUDIO_OK in case of success, AUDIO_ERROR otherwise
+*/
+uint8_t MP34DT01_Resume(void)
+{
+  int32_t counter = 0;
+  for (counter = MP34DT01_Handler.MicChannels; counter > 0; counter --)
+  {
+    if (HAL_OK != HAL_DFSDM_FilterRegularStart_DMA(&haudio_in_dfsdmfilter[counter-1],
+                                                   (int32_t*) RecBuff[counter-1],
+                                                   MP34DT01_Handler.Sampling_Freq / 1000 * 2))
+    {
+      return AUDIO_ERROR;
+    }
+  }
+  return AUDIO_OK;
+}
+
+
+/**
+* @brief  Controls the audio in volume level.
+* @param  Volume: Volume level to be set. This value has the same behaviour of the
+Volume parameter of the PDM to PCM software decimation library. Other
+strategies are possible in order to control the volume, for example
+to act on the right bit shift amount of the DFSDM peripheral
+Values must be in the range from 0 to 64
+* @retval AUDIO_OK in case of success, AUDIO_ERROR otherwise
+*/
+uint8_t MP34DT01_SetVolume(uint8_t Volume)
+{
+  AudioInVolume = Volume;
+  return AUDIO_OK;
+}
+
+/**
+* @brief  Audio IN Error callback function
+* @param  None
+* @retval None
+*/
+__weak void MP34DT01_Error_Callback(void)
+{
+  /* This function is called when an Interrupt due to transfer error on or peripheral
+  error occurs. */
+}
+
+/**
+* @}
+*/
+
+/** @defgroup MP34DT01_L4_DRIVER_AUDIO_L4_Private_Functions
+* @{
+*/
+
+/******************************************************************************
+Static Functions
+*******************************************************************************/
+
+/**
+* @brief  Initializes the Digital Filter for Sigma-Delta Modulators interface (DFSDM).
+* @param  AudioFreq: Audio frequency to be used to set correctly the DFSDM peripheral.
+* @retval AUDIO_OK if correct communication, else wrong communication
+*/
+static uint8_t DFSDMx_Init(uint32_t AudioFreq, uint32_t ChnlNbr) {
+
+  uint16_t shift_amount = 0;
+  uint32_t SincOrder = 0;
+  if(OverSampling(AudioFreq) == 32)
+  {
+    shift_amount = 5;
+    SincOrder = DFSDM_FILTER_SINC5_ORDER;
+  }
+  else if(OverSampling(AudioFreq) == 128)
+  {
+    shift_amount = 8;
+    SincOrder = DFSDM_FILTER_SINC4_ORDER;
+  }
+  else if(OverSampling(AudioFreq) == 64)
+  {
+    shift_amount = 10;
+    SincOrder = DFSDM_FILTER_SINC5_ORDER;
+  }
+
+  /*####AUDIO CH 1: DFSDM CHANNEL 2####*/
+  haudio_in_dfsdmchannel[0].Init.OutputClock.Activation   = ENABLE;
+  haudio_in_dfsdmchannel[0].Init.OutputClock.Selection    = DFSDM_CHANNEL_OUTPUT_CLOCK_AUDIO;

```

Comment:
Same question, is it valid for all series? Seems not: F7/F4 and L4 only

Reply:
This library is only available for board with DFSDM module => F4/F7/L4 series. But not tested on F4/F7!